static/js/message_fetch.js

```diff
@@ -37,16 +36,6 @@ function process_result(data, opts) {
         ui_report.hide_error($("#connection-error"));
     }
 
-    if (
-        messages.length === 0 &&
-        message_lists.current === message_list.narrowed &&
-        message_list.narrowed.empty()
-    ) {
-        // Even after trying to load more messages, we have no
-        // messages to display in this narrow.
-        narrow_banner.show_empty_narrow_message();
-    }
```

Comment:
I think that the correct fix probably involves replacing this code with, at the end of this function, checking if `current_msg_list.data.empty()`, instead of the current logic which checks whether the API endpoint returned messages (which might be in muted topics).

In other words, I think we can delete the `messages.length === 0` check, but only if we move this block after the `message_util.add_old_messages` calls.

Reply:
I had thought about moving that check instead of deleting it, so this makes sense to me. Updated to have the check for the empty narrowed / current list to be after `message_util.add_old_messages`.

Reply:
Would you like me to take a stab at that documentation / comment update? I'm assuming it would just be comments for those two variables.

Reply:
Probably the ideal thing to do would be for us to add comments for all the attributes of MessageListData/MessageList/MessageListView; I think they're an important enough part of the implementation that it'd be helpful to define them all. Possibly I should just do that in the post-release quiet this afternoon.